packages/graphql/src/translate/queryAST/utils/create-cypher-subquery.ts

```diff
@@ -60,21 +87,30 @@ function getProjectionExpr({
     fromVariable,
     fields,
     attribute,
+    nestedFields,
 }: {
     fromVariable: Cypher.Variable;
     fields: Record<string, string> | undefined;
     attribute: AttributeAdapter;
+    nestedFields?: Field[];
 }): Cypher.Expr {
     let projection: Cypher.Expr = fromVariable;
     if (fields) {
         projection = new Cypher.MapProjection(fromVariable);
-
-        for (const [alias, name] of Object.entries(fields)) {
-            if (alias === name) projection.set(alias);
-            else {
-                projection.set({
-                    [alias]: fromVariable.property(name),
-                });
+        // nested fields are presents only if the attribute is an object or an abstract type, and they produce a different projection
+        if (nestedFields) {
+            const subqueriesProjection = nestedFields?.map((f) => f.getProjectionField(fromVariable));
+            for (const subqueryProjection of subqueriesProjection) {
+                projection.set(subqueryProjection);
+            }
+        } else {
+            for (const [alias, name] of Object.entries(fields)) {
+                if (alias === name) projection.set(alias);
+                else {
+                    projection.set({
+                        [alias]: fromVariable.property(name),
+                    });
+                }
             }
```

Comment:
Maybe we could consider moving the conditionals of this logic to separate functions, to reduce complexity of this function.